### PR TITLE
REBASED Pr81x l1t-tsg-v9 (part 3) for Sept 2016 MC production

### PR DESCRIPTION
**Pull request by rekovic**
This is the l1t-tsg-v9 part 3 rebased version, i.e. rebased version of #15732.

It is rebased on CMSSW_8_1_X after part 2 (#15757) was merged.

### CODE REVIEW

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
please test

File: EventFilter/L1TRawToDigi/src/implementations_stage2/GMTCollections.h [hunk 1]
```diff
@@ -16,20 +16,35 @@ namespace l1t {
...
 
          private:
-            std::unique_ptr<RegionalMuonCandBxCollection> regionalMuonCandsBMTF_;
-            std::unique_ptr<RegionalMuonCandBxCollection> regionalMuonCandsOMTF_;
-            std::unique_ptr<RegionalMuonCandBxCollection> regionalMuonCandsEMTF_;
-            std::unique_ptr<MuonBxCollection> muons_;
+            std::auto_ptr<RegionalMuonCandBxCollection> regionalMuonCandsBMTF_;
+            std::auto_ptr<RegionalMuonCandBxCollection> regionalMuonCandsOMTF_;
+            std::auto_ptr<RegionalMuonCandBxCollection> regionalMuonCandsEMTF_;
+            std::auto_ptr<MuonBxCollection> muons_;
```
Comment (davidlange6) [inline EventFilter/L1TRawToDigi/src/implementations_stage2/GMTCollections.h]: auto_ptr needs to be migrated to unique_ptr.
@rekovic

File: DataFormats/L1TGlobal/interface/GlobalObject.h [hunk 2]
```diff
@@ -23,11 +23,12 @@ enum GlobalObject
     gtETT,
     gtHTT,
     gtHTM,
-    gtETM2,
+    gtETMHF,
```
Comment (davidlange6) [inline DataFormats/L1TGlobal/interface/GlobalObject.h]: @rekovic - this is a new name for the same thing? Or does this change introduce a backwards incompatibility (well, if we store this data format someplace)

Comment (rekovic) [inline DataFormats/L1TGlobal/interface/GlobalObject.h]: @davidlange6 - Yes, this is a new name for the same thing

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
please test

**Pull request merged**